### PR TITLE
grafana: Introduce additional groupby

### PR DESCRIPTION
**Pull request by Connor1996**
### What is changed and how it works?

Issue Number: Ref #15990 

What's Changed:

```commit-message
Introduce additional_groupby variable for grafana board and make instance variable multi-value
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

By default, `additional groupby` is set to `none` indicating it is disabled
b497c99d-0dcc-4be0-a602-07378ccfb725
When selecting `instance` for `additional groupby`, by instance are automatically performed
2b6e96c5-c9ed-4901-a0a8-86d81647f869

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Introduce additional groupby for tikv-details grafana board
```

### CODE REVIEW

File: metrics/grafana/common.py [hunk 1]
```diff
@@ -592,14 +627,27 @@ def expr_histogram_avg(
 
 
 def target(
-    expr: Union[Expr, str],
+    expr: Union[Expr, OpExpr, str],
     legend_format: Optional[str] = None,
     hide=False,
     data_source=DATASOURCE,
     interval_factor=1,  # Prefer "high" resolution
+    additional_groupby=False,
```
Comment (overvenus) [inline metrics/grafana/common.py:635]:
```suggestion
    additional_groupby: list[str] = [ADDITIONAL_GROUPBY],
```

It looks more natural to me if it's a list of string (labels).

**Review by LykxSassinator — COMMENTED** [review]
LGTM

**Review by LykxSassinator — APPROVED** [review]

File: metrics/grafana/common.py [hunk 2]
```diff
@@ -483,8 +516,10 @@ def expr_simple(
     return expr
 
 
-def expr_operator(lhs: Union[Expr, str], operator: str, rhs: Union[Expr, str]) -> str:
-    return f"""({lhs} {operator} {rhs})"""
+def expr_operator(
+    lhs: Union[Expr, OpExpr, str], operator: str, rhs: Union[Expr, OpExpr, str]
+) -> str:
```
Comment (hbisheng) [inline metrics/grafana/common.py:521]:
```suggestion
) -> OpExpr:
```

File: metrics/grafana/common.py [hunk 3]
```diff
@@ -592,14 +627,32 @@ def expr_histogram_avg(
...
     legend_format: Optional[str] = None,
     hide=False,
     data_source=DATASOURCE,
     interval_factor=1,  # Prefer "high" resolution
+    # It indicates whether to add additional groupby label to the groupby and legend.
+    # If the expr is already groupby by instance, no need to set it to True.
+    # Additional groupby is used to support optional by instance. By default, it doesn't
+    # take effect. When the variable $additional_groupby of dashboard is set to `instance`.
+    # The expr would be groupby by `instance`.
+    additional_groupby=False,
```
Comment (overvenus) [inline metrics/grafana/common.py:640]: How about enable it by default?

Comment (Connor1996) [reply metrics/grafana/common.py:640]: That would make the query too heavy. For the queries already being by instance, `additional_groupby=false` doesn't affect them.

**Review by overvenus — APPROVED** [review]

**Pull request merged**